lib/BoundaryClickWatcher/index.js

```diff
@@ -67,10 +75,13 @@ class BoundaryClickWatcher extends Component {
         ref={el => {
           this.container = el;
         }}
-        onClick={this.handleInitialClick}
-        onKeyPress={this.handleInitialClick}
+        onClick={this.handleInnerClick}
+        onKeyPress={this.handleInnerClick}
       >
-        {this.props.children}
+        {typeof this.props.children === 'function'
+          ? this.props.children(this.state.boundaryIsActive)
+          : this.props.children
```

Comment:
Is this so we can use this component as a means to trigger `initialClickHandler` & `outsideClickHandler` without passing the active state to the child?